quickemu

```diff
@@ -933,7 +933,12 @@ function vm_boot() {
 }
 
 function shortcut_create {
+  local dirname="${HOME}/.local/share/applications"
   local filename="${HOME}/.local/share/applications/${VMNAME}.desktop"
```

Comment:
To keep in line with DRY, I'd suggest using the new `dirname` variable here as the base, and then adding `/${VMNAME}.desktop` to it to get the final `filename`.

Reply:
Good suggestion.  I'm in two minds though: for me this is a little too far in the direction of premature refactoring since

- it doesn't really pass the [rule of three](https://en.wikipedia.org/wiki/Rule_of_three_(computer_programming))
- it makes the code slightly less obvious without any significant performance or security benefit
- it impacts code beyond the minimal scope of this patch.

I know I am overstating my counter-argument here (and would be happy to have a chat over a beer in [![Discord](https://img.shields.io/discord/712850672223125565?color=0C306A&label=WimpysWorld%20Discord&logo=Discord&logoColor=ffffff&style=flat-square)](https://discord.gg/sNmz3uw) ), but my preference would be to consider this refactoring along with further possible DRY-cleaning in a separate issue/PR.

Reply:
I wouldn't call it _refactoring_ in this case. The `dirname` variable is only introduced in this PR and thus this is the first place where we _can_ use it to then compose it into the `filename`. 

Anyways, they're just my two cents. Feel free to ignore me if you disagree.